request.go

```diff
@@ -381,73 +150,54 @@ func (r *request) ConvertOptions() error {
 	return nil
 }
 
-// NewEmptyRequest initializes an empty request
-func NewEmptyRequest() (Request, error) {
-	return NewRequest(nil, nil, nil, nil, nil, nil)
-}
-
-// NewRequest returns a request initialized with given arguments
-// An non-nil error will be returned if the provided option values are invalid
-func NewRequest(path []string, opts cmdkit.OptMap, args []string, file files.File, cmd *Command, optDefs map[string]cmdkit.Option) (Request, error) {
-	if opts == nil {
-		opts = make(cmdkit.OptMap)
-	}
-	if optDefs == nil {
-		optDefs = make(map[string]cmdkit.Option)
+// GetEncoding returns the EncodingType set in a request, falling back to JSON
+func GetEncoding(req *Request) EncodingType {
+	encIface := req.Options[EncLong]
+	if encIface == nil {
+		return JSON
 	}
 
-	ctx := Context{}
-	values := make(map[string]interface{})
-	req := &request{
-		path:       path,
-		options:    opts,
-		arguments:  args,
-		files:      file,
-		cmd:        cmd,
-		ctx:        ctx,
-		optionDefs: optDefs,
-		values:     values,
-		stdin:      os.Stdin,
+	switch enc := encIface.(type) {
```

Comment:
seems a bit hacky, whats the issue here? something about marshaling back and forth not preserving string types?

Reply:
yes, cmdkit.Option does not know about types, only about reflect.Kind (and that is reflect.String for both type string and EncodingType). I think in the long run cmdkit should be taught to handle types properly, especially for that strings PR that has been out there for a while now. It assumes "if kind is Array, it's got to be a string array!" which is wrong imho, but doing it right would require a lot of changes and it didn't have priority.